### DISCUSSION TITLE
Calendar not working on angular

### DISCUSSION THREAD

**Discussion by PigeonsPersonified** [add2cal/add-to-calendar-button#481] [Q&A] [answered]
Added this package and followed the steps to set this up as part of my project. Im trying to add this calendar button as a menu item however it asks for an ics file even though i only want to use outlook, microsoft teams and microsoft365, not sure why it wants one but i couldnt find a workaround.

**Comment by jekuer** [comment]
Can you please share the code snippet you are using?

**Reply by PigeonsPersonified** [reply]
```
<add-to-calendar-button
          mat-menu-item
          name="Incase365:"
          options="'Microsoft365', 'Outlook.com', 'MicrosoftTeams'"
          location="World Wide Web"
          startDate="2023-10-30"
          endDate="2023-10-30"
          startTime="12:00"
          endTime="12:00"
          timeZone="Europe/London"
          debug
        ></add-to-calendar-button>
```

**Reply by jekuer** [reply]
Thanks a lot!
What exactly does the debug message say and which version are you using?

**Reply by PigeonsPersonified** [reply]
using the latest version (2.4.3), the error returns a follows 
![Screenshot 2023-10-09 144007](https://github.com/add2cal/add-to-calendar-button/assets/147399447/41a3c2a2-5454-433f-b6cd-1eb9602b0b91)

**Reply by jekuer** [reply]
Please check my Stackblitz demo at https://stackblitz.com/edit/add-to-calendar-button-angular-vncxay?file=src%2Fapp%2Fapp.component.html

Are you sure the above is the code you are using and that you are also not dynamically changing any attributes?

**Reply by PigeonsPersonified** [reply]
Yea, the end game plan is to have it dynamic but i wanted to set it up with no dynamic attributes just to make sure it works and whatnot. Its setup exactly the same as the demo so im not sure where its going wrong.

**Reply by jekuer** [reply]
It fails at https://github.com/add2cal/add-to-calendar-button/blob/main/src/atcb-validate.js#L124
subscribe can be only true if it is at least somehow setup as attribute. The above code has nothing of this word in it. So, if this is exactly the code you are using, in this plain way, it will be hard to debug from here on.

The only thing I could recommend is to setup a stackblitz or codepen or similar, so other can have a look.

**Reply by PigeonsPersonified** [reply]
ill get on that now, bit of a tangent but are you aware of any packages or angular features that could conflict with it, seems like the most logical reason for the subscribe to be affected

**Reply by jekuer** [reply]
No really.
Could you maybe inspect the HTML in your browser and check how the <add-to-calendar-button /> tag looks like?

**Reply by PigeonsPersonified** [reply]
![image](https://github.com/add2cal/add-to-calendar-button/assets/147399447/ebb0d6be-417e-4f39-bdcf-7a5b85a15d81)
Is this what you meant?

**Reply by jekuer** [reply, marked as answer]
Yes. This is really strange. Does it work if you remove the "mat-menu-item". This adds a lot of additional stuff to the component.

**Reply by PigeonsPersonified** [reply]
Ahh finally! That seems to have worked. Not sure why but removing the mat menu item worked, however now i cant have it where i wanted, should be ok as long as i can customise it. Thank you man :)

**Reply by jekuer** [reply]
Put it in a wrapping div with that attribute. This should help.

**Reply by PigeonsPersonified** [reply]
Thanks again man. appreciate the support. Have a nice day
